qa/os/src/test/java/org/elasticsearch/packaging/test/KeystoreManagementTests.java

```diff
@@ -80,9 +83,30 @@ public void test11InstallPackageDistribution() throws Exception {
         Shell.Result r = sh.runIgnoreExitCode(bin.keystoreTool.toString() + " has-passwd");
         assertThat("has-passwd should fail", r.exitCode, not(is(0)));
         assertThat("has-passwd should fail", r.stderr, containsString("ERROR: Keystore is not password-protected"));
+        Shell.Result r2 = bin.keystoreTool.run("list");
+        assertThat(r2.stdout, containsString("keystore.seed"));
+    }
+
+    /** Test initial Docker state */
+    public void test12InstallDockerDistribution() throws Exception {
+        assumeTrue(distribution().packaging == Distribution.Packaging.DOCKER);
+
+        installation = Docker.runContainer(distribution());
+
+        try {
+            waitForPathToExist(installation.config("elasticsearch.keystore"));
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        final Installation.Executables bin = installation.executables();
+        Shell.Result r = sh.runIgnoreExitCode(bin.keystoreTool.toString() + " has-passwd");
+        assertThat("has-passwd should fail", r.exitCode, not(is(0)));
+        assertThat("has-passwd should fail", r.stdout, containsString("ERROR: Keystore is not password-protected"));
```

Comment:
Since the previous line checked that the command failed, how about changing the message here:

```suggestion
        assertThat("has-passwd didn't fail with the expected error", r.stdout, containsString("ERROR: Keystore is not password-protected"));
```